packages/core/src/controller/createOptimistic.ts

```diff
@@ -0,0 +1,53 @@
+import { OPTIMISTIC_TYPE } from '@rest-hooks/core/actionTypes';
+import type { EndpointInterface } from '@rest-hooks/endpoint';
+import { EndpointUpdateFunction } from '@rest-hooks/core/controller/types';
+
+import { OptimisticAction } from '..';
+
+export default function createOptimistic<
+  E extends EndpointInterface & {
+    update?: EndpointUpdateFunction<E>;
+  },
+>(
+  endpoint: E,
+  {
+    args,
+  }: {
+    args: readonly [...Parameters<E>];
+  },
+): OptimisticAction {
+  const expiryLength: number = endpoint.dataExpiryLength ?? 1000;
+  /* istanbul ignore next */
+  if (process.env.NODE_ENV === 'development' && expiryLength < 0) {
+    throw new Error('Negative expiry length are not allowed.');
```

Comment:
so what happens in production if you pass a negative length? there’s nothing that stops someone from relying on that - sometimes people have NODE_ENV always set to production, even in development.

Reply:
The dev only checks are all to help discover poorly written static code - as a supplement to typescript as there are things it cannot check. Running local dev with prod means you're opting out of that, just like someone can opt out of typescript.

Reply:
I guess I’m wondering why provide less protection at runtime, rather than always throwing this error?

Reply:
Smaller runtime, faster execution time.

Reply:
Seems far less important than correctness to me, but ok

Reply:
The constraint problem is defined as:

- maximize performance and minimize client payload
- ensure catching all static definition errors as early and easy as possible
- guarantee no catastrophic failures in production via isolation

Even with a bad definition here, the result is not catastrophic failure